### ISSUE TITLE
NullReferenceException when cache key not found, when non-generic cache policy executing for non-nullable types

### ISSUE THREAD

**Issue by cmeeren** [App-vNext/Polly#472]
When defining a cache policy using the following code:

```f#
open Microsoft.Extensions.Caching.Memory
open Polly
open Polly.Caching.Memory

let cachePolicy =
  Policy.Cache(
    MemoryCacheProvider(new MemoryCache(MemoryCacheOptions())),
    TimeSpan.FromHours 24.,
    fun ctx key ex ->
      Log.Error(ex, sprintf "Cache provider threw exception for key %s" key)
  )
```

the `onCacheError` action is triggered when the item does not exist in cache, with the following exception:

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at TCacheFormat Polly.Caching.GenericCacheProvider<TCacheFormat>.Polly.Caching.ISyncCacheProvider<TCacheFormat>.Get(?)
   at TResult Polly.Caching.CacheEngine.Implementation<TResult>(ISyncCacheProvider<TResult> cacheProvider, ITtlStrategy<TResult> ttlStrategy, Func<Context, string> cacheKeyStrategy, Func<Context, CancellationToken, TResult> action, Context context, CancellationToken cancellationToken, Action<Context, string> onCacheGet, Action<Context, string> onCacheMiss, Action<Context, string> onCachePut, Action<Context, string, Exception> onCacheGetError, Action<Context, string, Exception> onCachePutError)
```

**Comment by reisenberger** [comment]
@cmeeren. Thank-you for raising this. What is the return type of the delegate being executed through the policy? (ie the type of `TResult` or `TCacheFormat`).  Could you post also the code placing the execution through the policy?  If my assumptions are correct, fix should be quick and easy.

**Comment by cmeeren** [comment]
```f#
let getCompanyName (signedInUser: User) (userToGetFor: User) : Skippable<string option> =
  cachePolicy.Execute(
    (fun _ -> getUncached signedInUser userToGetFor),
    Context(string userToGetFor.CompanyID)
  )
```

Where `Skippable` is an F# union type I've made that's similar to `option`.

**Comment by reisenberger** [comment]
Thank you @cmeeren for raising this. Fix is already coded and tested [here](https://github.com/reisenberger/Polly/commit/1197781a563db37fe1f1e8980950db88b5257d53). We should be able to release this early next week.

**Comment by reisenberger** [comment]
Thank you @cmeeren for raising this! Fixed in v6.1.0 which will release to nuget shortly
